state-synchronizer/src/chunk_request.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Error, Result};
+use libra_types::crypto_proxies::LedgerInfoWithSignatures;
+use libra_types::transaction::Version;
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+use std::fmt;
+
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq)]
+pub enum TargetType {
+    /// The response is built relative to the target (or end of epoch).
+    TargetLedgerInfo(LedgerInfoWithSignatures),
+    /// The response is built relative to the highest available LedgerInfo (or end of epoch).
+    /// The value specifies the timeout in ms to wait for an available response.
+    HighestAvailable { timeout_ms: u64 },
```

Comment:
why do we send a timeout as part of our request? Why would our peer care?

Reply:
It's for the long polling requests of the full nodes (I was trying to explain it in the comments, please tell if the comments are confusing). In case the information is not immediately available the upstream peer might keep the request as a subscriber until the information becomes available within the next `timeout_ms`.

Reply:
I guess a doc block on the TargetType might be useful.

It might also be useful to state "long poll" in this doc block.